File: tests/enums/with-simple-data.js
export function makeEmpty() {
  return [0];
}
export function makeUpc$numberSystem$manufacturer$product$check$(numberSystem, manufacturer, product, check) {
  return [1, numberSystem, manufacturer, product, check];
}
export function makeQr$value$(value) {
  return [2, value];
}
export function describe$barcode$(barcode) {
  var $match = barcode;

  if ($match[0] === 0) {
    return "Empty";
  } else if ($match[0] === 1) {
    const numberSystem = $match[1];
    const manufacturer = $match[2];
    const product = $match[3];
    const check = $match[4];
    return "UPC:" + String(numberSystem) + "-" + String(manufacturer) + "-" + String(product) + "-" + String(check);
  } else if ($match[0] === 2) {
    const value = $match[1];
    return "QR:" + value;
  }
}

File: tests/enums/with-simple-data.swift
public enum Barcode {
	case empty
    case upc(Int)
    case qrCode(String)
}

public func make(empty: ()) -> Barcode {
	return .empty
}

public func make(upc value: Int) -> Barcode {
	return .upc(value)
}

public func make(qrCode value: String) -> Barcode {
	return .qrCode(value)
}

public func describe(barcode: Barcode) -> String {
	switch barcode {
		case .empty:
			return "Empty"
		case .upc(let value):
			return "UPC:" + String(value)
		case .qrCode(let value):
			return "QR:" + value
	}
}
